src/components/SocialMediaLinks/SocialMediaLinks.tsx

```diff
@@ -0,0 +1,209 @@
+import {
+  chakra,
+  useMultiStyleConfig,
+  // useColorModeValue,
+} from "@chakra-ui/react";
+import List from "../List/List";
+import Link from "../Link/Link";
+import Icon, { IconSizes } from "../Icons/Icon";
+import { LayoutTypes } from "../../helpers/types";
+import { socialMediaDataMap } from "./SocialMediaDataMap";
+import React, { forwardRef } from "react";
+
+export const borderTypeArray = ["none", "circular", "straight"] as const;
+export type BorderType = typeof borderTypeArray[number];
+
+export const colorTypeArray = ["link", "textDefault", "textInverse"] as const;
+export type ColorType = typeof colorTypeArray[number];
+
+export const sizeTypeArray = ["small", "medium", "large"] as const;
+export type SizeType = typeof sizeTypeArray[number];
+
+// @todo The below export fails TS, for reasons I do not yet understand. But it would be useful so that we don't
+//   need to update in two places if any names are changed / added / subtracted.
+//   export const socialMediaLinkTypeArray  = socialMediaDataMap.map(({ type }) => type) as const;
+export const socialMediaLinkTypeArray = [
+  "blogs",
+  "facebook",
+  "instagram",
+  "pinterest",
+  "soundcloud",
+  "tiktok",
+  "tumbler",
+  "twitter",
+  "youtube",
+] as const;
+export type SocialMediaLinkType = typeof socialMediaLinkTypeArray[number];
+
+export interface SocialMediaLinkDataProps {
+  /** Optional override for default platform name */
+  labelText?: string;
+  /** Required. Must be one of socialMediaLinkTypeArray */
+  type: SocialMediaLinkType;
+  /** Optional override for default social media url */
+  url?: string;
+}
+
+export interface SocialMediaLinksProps {
+  /** Optional border: straight, circular or none. */
+  borders?: BorderType;
+  /** Optional className you can add in addition to "social-media-links." */
+  className?: string;
+  /** Optional color */
+  color?: ColorType;
+  /** ID that other components can cross-reference for accessibility purposes */
+  id?: string;
+  /** Optional layout */
+  layout?: LayoutTypes;
+  /** Optional array of social media platform types, urls, and label texts */
+  linksData?: SocialMediaLinkDataProps[];
+  /** Optional true/false to display names of platform along with icon
```

Comment:
Nitpick: add a period at the end of "icon" because when it gets rendered in Storybook, it's confusing to read.